src/stores/templateStore.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const _ = require('lodash');
+const chromeApi = require('chromeback')(chrome);
+
+const alt = require('../alt');
+const client = require('../lib/client');
+const getErrorMessage = require('../lib/getErrorMessage');
+const { fetchNewTemplate, submitTemplate } = require('../actions/syncTemplateActions');
+const { cancelChanges, setDeltaTemplate, setDeltaUrl } = require('../actions/editTemplateActions');
+
+class TemplateStore {
+  constructor(){
+    this.state = {
+      prTemplate: null,
+      templateUrl: null,
+      errorMessage: null,
+      deltaUrl: null,
+      deltaTemplate: null,
+      disableCancel: true,
+      disableSubmit: true
+    };
+
+    this.bindListeners({
+      handleCancelChanges: cancelChanges,
+      handleFetchNewTemplate: fetchNewTemplate,
+      handleSetDeltaTemplate: setDeltaTemplate,
+      handleSetDeltaUrl: setDeltaUrl,
+      handleSubmitTemplate: submitTemplate
+    });
+
+    chromeApi.storage.sync.get(['templateUrl', 'prTemplate'], (err, res)=>{
+      if(err){
+        this.setState({ errorMessage: getErrorMessage(err) });
+      } else {
+        this.setState({
+          prTemplate: res.prTemplate,
+          templateUrl: res.templateUrl,
+          deltaUrl: res.templateUrl,
+          deltaTemplate: res.prTemplate
+        });
+      }
+    });
+  }
+
+  handleCancelChanges(){
+    this.setState({
+      deltaUrl: this.state.templateUrl,
+      deltaTemplate: this.state.prTemplate,
+      disableCancel: true,
+      disableSubmit: true,
+      errorMessage: ''
+    });
+  }
+
+  handleFetchNewTemplate(){
+    const { deltaUrl, prTemplate, templateUrl } = this.state;
+    client(deltaUrl)
+      .then(({ entity })=>{
+        this.setState({
+          deltaTemplate: entity,
+          errorMessage: ''
+        });
+        if(entity === prTemplate){
+          this.setState({ disableSubmit: true });
+          if(deltaUrl === templateUrl){
+            this.setState({ disableCancel: true });
+          }
+        } else {
+          this.setState({
+            disableSubmit: false,
+            disableCancel: false
+          });
+        }
+      })
+      .otherwise((err)=>{
+        this.setState({
+          errorMessage: getErrorMessage(err),
+          disableSubmit: true
+        });
+      });
+  }
+
+  handleSetDeltaTemplate(newTemplate){
+    this.setState({
+      deltaTemplate: newTemplate.target.value,
```

Comment:
i'm a little wary about `newTemplate` being an event object that was dispatched, I think the `value` should be pulled out inside the component and the action called with it as the param.  That allows you to do `evt.preventDefault`, etc where the event handler is actually registered